test/checkChallenge.test.js

```diff
@@ -0,0 +1,1627 @@
+
+    });
+
+    it('accept handicap edge max', () => {
+
+      const notification = base_challenge({ ranked: false, handicap: 6 });
+
+      config.minhandicap = 0;
+      config.maxhandicap = 6;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+
+    });
+
+    it('accept handicap edge max (fakerank automatic handicap stone number estimation)', () => {
+
+      const notification = base_challenge({ ranked: false, user: { ranking: 30 }, handicap: -1 }); // "1d"
+
+      config.fakerank    = 24; // "6k"
+      // "1d" - "6k" = 30 - 24 = 6 automatic handicap stones
+      config.minhandicap =  0;
+      config.maxhandicap =  6;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+
+    });
+
+    it('reject handicap too high (even games only) ', () => {
+
+      const notification = base_challenge({ ranked: false, handicap: 1 });
+
+      config.minhandicap = 0;
+      config.maxhandicap = 0;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Maximum number of handicap stones is 0, please reduce the number of handicap stones (no handicap games).' }));
+
+    });
+
+    it('reject handicap too high ', () => {
+
+      const notification = base_challenge({ ranked: false, handicap: 9 });
+
+      config.minhandicap = 0;
+      config.maxhandicap = 6;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Maximum number of handicap stones is 6, please reduce the number of handicap stones.' }));
+
+    });
+
+    it('reject handicap too high (fakerank automatic handicap stone number estimation)', () => {
+
+      const notification = base_challenge({ ranked: false, user: { ranking: 17 }, handicap: -1 }); // "13k"
+
+      config.fakerank    = 32; // "3d"
+      // "3d" - "13k" = 32 - 17 = 15 automatic handicap stones
+      config.minhandicap = 0;
+      config.maxhandicap = 6;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Maximum number of handicap stones is 6, please reduce the number of handicap stones, please manually select the number of handicap stones in -custom handicap-.' }));
+
+    });
+
+  });
+
+  describe('Byoyomi time settings', () => {
+
+    // sample:
+    // {"system":"byoyomi","time_control":"byoyomi","speed":"live","pause_on_weekends":false,"main_time":1200,"period_time":30,"periods":5}
+
+    // Main Time Blitz
+
+    it('reject main time blitz too low', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "byoyomi", time_control: "byoyomi", speed: "blitz", main_time: 1, periods: 1, period_time: 1 } });
+
+      config.minmaintimeblitz = 10;
+      config.maxmaintimeblitz = 30;
```

Comment:
actually you're right, we should also test a case where only min and only max arg are used because we never know if it would behave the same as the 2 args used at the same time (but it would be overkill maybe so we may not do it in this pr which is already very long)